### ISSUE TITLE
Loading large RichtTextLabel takes a long time

### ISSUE THREAD

**Issue by maran** [godotengine/godot#96475]
### Tested versions

4.3.stable

### System information

Pop!_OS 22.04

### Issue description

My app starting more slowly with every version I released. I've been debugging it for most of the day only to find out that my change-log was the culprit. I've been using a RichtTextLabel and just added to this label with every release. Right now loading the scene which includes the release notes takes about 6 seconds on my 11th gen i7. 

### Steps to reproduce

1. Add a RichtTextLabel to a scene.
2. Create a nice Lorem Ipsum and copy it a few dozen times into this RichTextLabel.
3. See your scene loading slower with every paragraph added.

### Minimal reproduction project (MRP)

[RichTextLabel.zip](https://github.com/user-attachments/files/16838332/RichTextLabel.zip)

**Comment by RedMser** [comment]
By turning on the `Threaded` checkbox in your MRP, the scene loads instantly

![image](https://github.com/user-attachments/assets/f7ff5c9f-cae9-44fa-8446-526f499cc8e9)

**Comment by RedMser** [comment]
> Any downside to using threaded?

None are documented, [the PR](https://github.com/godotengine/godot/pull/61142) only says that it is off by default but not why...

**Comment by aXu-AP** [comment]
> Any downside to using threaded?

It's a little hidden in the docs under [bbcode tutorial](https://docs.godotengine.org/en/4.3/tutorials/ui/bbcode_in_richtextlabel.html#performance), but it says:

> Enable the Threading > Threaded property in RichTextLabel. This won't speed up processing, but it will prevent the main thread from blocking, which avoids stuttering during gameplay. Only enable threading if it's actually needed in your project, as threading has some overhead.

So I guess if you have multiple smaller RichTextLabels, it can affect performance negatively. For long texts I think it's best to have them run on background.

**Comment by Mickeon** [comment]
> Also the reason why it's disabled by default might be in that the behavior is easier to understand when it's blocking.

Correct. You'd expect the label to be changed relatively instantly. By enabling `threaded` you should expect at least one frame where the label appears invisible or only partially generated, which is more noticeable the longer `text` is and is most definitely not desired unless you're generating a web page's worth of text. In OP's case, that's exactly what `threaded` is for.

**Comment by Grublady** [comment]
CC @AThousandShips

Confirmed reproducible (by others in this thread and myself)—
Should be OK to remove Needs Testing label.

**Comment by Kaleb-Reid** [comment]
I'm pretty sure that this is the reason why loading large diffs in the Version Control tab takes forever, so I did some profiling and at least when using the advanced text server the vast majority of the time is spent here: https://github.com/godotengine/godot/blob/16a11ac88b3aedac2825ec570910f34fb40f3f98/modules/text_server_adv/text_server_adv.cpp#L5297-L5298 I messed around with it and the most time by a large margin is always spent in the line first accessing ```sd_glyphs[j]```.
